library/src/com/google/maps/android/kml/KmlLayer.java

```diff
@@ -0,0 +1,147 @@
+package com.google.maps.android.kml;
+
+import com.google.android.gms.maps.GoogleMap;
+
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+import org.xmlpull.v1.XmlPullParserFactory;
+
+import android.content.Context;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Document class allows for users to input their KML data and output it onto the map
+ */
+public class KmlLayer {
+
+    private final KmlRenderer mRenderer;
+
+    /**
+     * Creates a new KmlLayer object
+     *
+     * @param map        GoogleMap object
+     * @param resourceId Raw resource KML file
+     * @param context    Context object
+     * @throws XmlPullParserException if file cannot be parsed
+     */
+    public KmlLayer(GoogleMap map, int resourceId, Context context)
+            throws XmlPullParserException, IOException {
+        this(map, context.getResources().openRawResource(resourceId));
+    }
+
+    /**
+     * Creates a new KmlLayer object
+     *
+     * @param map    GoogleMap object
+     * @param stream InputStream containing KML file
+     * @throws XmlPullParserException if file cannot be parsed
+     */
+    public KmlLayer(GoogleMap map, InputStream stream)
+            throws XmlPullParserException, IOException {
+        mRenderer = new KmlRenderer(map);
+        XmlPullParser xmlPullParser = createXmlParser(stream);
+        KmlParser parser = new KmlParser(xmlPullParser);
+        parser.parseKml();
+        stream.close();
+        mRenderer.storeKmlData(parser.getStyles(), parser.getStyleMaps(), parser.getPlacemarks(),
+                parser.getContainers(), parser.getGroundOverlays());
+    }
+
+    /**
+     * Creates a new XmlPullParser to allow for the KML file to be parsed
+     *
+     * @param stream InputStream containing KML file
+     * @return XmlPullParser containing the KML file
+     * @throws XmlPullParserException if KML file cannot be parsed
+     */
+    private static XmlPullParser createXmlParser(InputStream stream) throws XmlPullParserException {
+        XmlPullParserFactory factory = XmlPullParserFactory.newInstance();
+        factory.setNamespaceAware(true);
+        XmlPullParser parser = factory.newPullParser();
+        parser.setInput(stream, null);
+        return parser;
+    }
+
+    /**
+     * Adds the KML data to the map
+     *
+     * @throws XmlPullParserException if KML file cannot be parsed
+     * @throws IOException            if KML file cannot be opened
+     */
+    public void addKmlData() throws IOException, XmlPullParserException {
```

Comment:
I'm torn as to whether we should rename this or not... Since you have the rendering separated from the constructor, I think the method name should reflect that rendering/drawing is happening. This sounds kinda ambiguous in that it mentions data is being added, but doesn't really say where (I know, it's in the docs, but it should be obvious from the method too).

I propose renaming this to draw(). Thoughts @lchans @juliawong @amorris13?